src/app/core/services/user.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { IUser } from '../../pages/user/user.model';
+import { Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Users } from '../../../assets/data/users';
+
+@Injectable()
+export class UserService {
+  currentUser: IUser;
```

Comment:
This field should be private. Access to him should be only through a getter/setter. 
Components should trigger login/logout/registration methods that will be set currentUser value